prediction_market_agent/agents/crewai_subsequential_agent/crewai_agent_subquestions.py

```diff
@@ -0,0 +1,192 @@
+        return Outcomes.model_validate_json(result)
+
+    def build_tasks_for_outcome(self, input_dict: dict[str, t.Any] = {}) -> list[Task]:
+        task_research_one_outcome = Task(
+            description=RESEARCH_OUTCOME_PROMPT.format(**input_dict),
+            agent=self.researcher,
+            expected_output=RESEARCH_OUTCOME_OUTPUT,
+            async_execution=True,
+        )
+        task_create_probability_for_one_outcome = Task(
+            description=PROBABILITY_FOR_ONE_OUTCOME_PROMPT,
+            expected_output=PROBABILITY_CLASS_OUTPUT,
+            agent=self.predictor,
+            output_json=ProbabilityOutput,
+            async_execution=False,
+            context=[task_research_one_outcome],
+        )
+
+        return [task_research_one_outcome, task_create_probability_for_one_outcome]
+
+    def generate_prediction_for_one_outcome(self, sentence: str) -> ProbabilityOutput:
+        task_research_one_outcome = Task(
+            description=RESEARCH_OUTCOME_PROMPT,
+            agent=self.researcher,
+            expected_output=RESEARCH_OUTCOME_OUTPUT,
+        )
+        task_create_probability_for_one_outcome = Task(
+            description=PROBABILITY_FOR_ONE_OUTCOME_PROMPT,
+            expected_output=PROBABILITY_CLASS_OUTPUT,
+            agent=self.predictor,
+            output_json=ProbabilityOutput,
+            context=[task_research_one_outcome],
+        )
+        crew = Crew(
+            agents=[self.researcher, self.predictor],
+            tasks=[task_research_one_outcome, task_create_probability_for_one_outcome],
+            verbose=2,
+            process=Process.sequential,
+        )
+
+        result = crew.kickoff(inputs={"sentence": sentence})
+        return ProbabilityOutput.model_validate_json(result)
+
+    def generate_final_decision(
+        self, outcomes_with_probabilities: list[t.Tuple[str, ProbabilityOutput]]
+    ) -> ProbabilityOutput:
+        task_final_decision = Task(
+            description=FINAL_DECISION_PROMPT,
+            agent=self.predictor,
+            expected_output=PROBABILITY_CLASS_OUTPUT,
+            output_json=ProbabilityOutput,
+        )
+
+        crew = Crew(
+            agents=[self.predictor],
+            tasks=[task_final_decision],
+            verbose=2,
+        )
+
+        crew.kickoff(
+            inputs={
+                "outcomes_with_probabilities": [
+                    (i[0], i[1].dict()) for i in outcomes_with_probabilities
+                ],
+                "number_of_outcomes": len(outcomes_with_probabilities),
+                "outcome_to_assess": outcomes_with_probabilities[0][0],
+            }
+        )
+        return ProbabilityOutput.model_validate_json(
+            task_final_decision.output.raw_output
+        )
+
+    def answer_binary_market(self, question: str) -> ProbabilityOutput:
+        outcomes = self.split_research_into_outcomes(question)
+        print("outcomes ", outcomes)
```

Comment:
```suggestion
        logger.debug(f"{outcomes=}"
```

I guess this should be a debug?

Reply:
Yep but no loguru currently in PMA :(

Reply:
Why not? https://github.com/search?q=repo%3Agnosis%2Fprediction-market-agent%20loguru&type=code